src/parse/parse_step.go

```diff
@@ -109,7 +109,7 @@ func checkSubrepo(state *core.BuildState, label, dependent core.BuildLabel, mode
 	}
 
 	// Try parsing the package in the host repo first.
-	s, err := parseSubrepoPackage(state, sl.PackageName, "", label, mode)
+	s, err := parseSubrepoPackage(state, sl.PackageName, sl.Subrepo, label, mode)
```

Comment:
sl.Subrepo is now set to the arch part i.e. for `///go@darwin_amd64//build_defs:go`, sl is `BuildLabel{PackageName: "go", Subrepo: "darwin_amd64"}`